### PR TITLE
Add hackathons section

### LINKED ISSUE #30: Hackathons > Container (desktop + mobile)
Add container to hold all relevant components

Add string "Subscribe to our newsletter to stay up to date on our hackathons!" underneath the section with the link directing to the newsletter subscribe pop-up - the on hover should prompt colour change in the text: 
Screen Shot 2021-08-04 at 6 51 07 PM

Screen Shot 2021-08-04 at 6 49 59 PM
Screen Shot 2021-08-04 at 6 50 12 PM

### PR DESCRIPTION
**Pull request by kevin-zou**
## Description
Resolves #30. This PR adds the hackathon section to the landing page, complete with photos and some placeholder text.

I refactored the styling in the `HackathonCard` component since it had some weird behaviour on hover. It would show the image for the last hackathon (ie. cmd-f) regardless of which hackathon you were hovering. This was probably because the selector on line 19 (in the original code) came back with 3 options (as there are 3 `CardContainer`s) and just gave back the last one by default (cmd-f).

### Stop using ThemeContext
I realized that I don't always need to use ThemeContext to fetch theme properties, and therefore I do not need to declare `styled` components within the component function. I think this has been messing up the styling when statically exporting the site with Next.js, so let's see if this fixes it. Regardless this cleans up the code!

## Other considerations
- refactored a bit of the Resources component after making some changes to the `LinkBody` typography component, @vincentchiang1 lmk if this is okay
- this doesn't have the star in the background, it was hard to implement and I think I went about it the wrong way - going to open an issue to come back to this and the other section backgrounds I've already done (#84)

### CODE REVIEW

File: components/Hackathons.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      align-items: center;
+    }
+  `;
+
+  return (
+    <>
+      <HackathonsContainer>
+        <HackathonCard
+          registrationOpen={false}
+          link="https://hackcamp.nwplus.io"
```
Comment (vincentchiang1) [inline components/Hackathons.js:27]: same with these link and file path strings, constants would be nice!

File: components/Hackathons.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          imageLink="/assets/nwHacks2021.png"
+        />
+        <HackathonCard
+          registrationOpen={false}
+          link="https://cmd-f.nwplus.io"
+          dateString="TBA"
+          imageLink="/assets/cmd-f2021.png"
+        />
+      </HackathonsContainer>
+      <Body align="center"><LinkBody onClick={() => setShowModal(true)}>Subscribe</LinkBody> to our newsletter to stay up to date on our hackathons!</Body>
```
Comment (vincentchiang1) [inline components/Hackathons.js:44]: nit: lets use constants for these string values for the sake of cleanliness :)

File: components/Hackathons.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        />
+        <HackathonCard
+          registrationOpen={false}
+          link="https://nwhacks.io"
+          dateString="TBA"
+          imageLink="/assets/nwHacks2021.png"
+        />
+        <HackathonCard
+          registrationOpen={false}
+          link="https://cmd-f.nwplus.io"
```
Comment (vincentchiang1) [inline components/Hackathons.js:39]: same with these link and file path strings, constants would be nice!

**Review by vincentchiang1 — APPROVED** [review]
aside from the nits with the constants, it looks superb!

File: components/Hackathons.js (hunk 2 above)
Comment (kevin-zou) [reply components/Hackathons.js]: Sounds good! Added here: fb50dd4

File: components/Hackathons.js (hunk 3 above)
Comment (kevin-zou) [reply components/Hackathons.js]: Done in fb50dd4 :)

**Review by panjenny0 — APPROVED** [review]
🙏🙏🙏👏👏👏👏

**Pull request merged**